test/integration/autonomous_upstream.cc

```diff
@@ -166,4 +169,16 @@ std::unique_ptr<Http::MetadataMapVector> AutonomousUpstream::preResponseHeadersM
   return std::move(pre_response_headers_metadata_);
 }
 
+AssertionResult AutonomousUpstream::closeConnection(uint32_t index,
+                                                    std::chrono::milliseconds timeout) {
+  if (!shared_connections_[index]->connected()) {
+    return AssertionSuccess();
+  }
```

Comment:
Technically can't this race? Should this part also be executed on the dispatcher and just ASSERT there?

Reply:
Sounds good.